core/src/test/java/com/klaytn/caver/scenario/RoleBasedAccountScenario.java

```diff
@@ -0,0 +1,396 @@
+                BRANDON.getAddress(),
+                address,
+                BigDecimal.valueOf(1),
+                Convert.Unit.KLAY, GAS_LIMIT
+        ).send();
+
+        assertEquals("0x1", transactionReceipt.getStatus());
+    }
+
+    protected KlayCredentials createAccount() throws Exception {
+        KlayCredentials user = KlayCredentials.create(Keys.createEcKeyPair());
+        chargeAccount(user.getAddress());
+        return user;
+    }
+
+    protected KlayCredentials createAccount(KlayCredentials oldCredentials) throws Exception {
+        ECKeyPair ecKeyPair = Keys.createEcKeyPair();
+        KlayCredentials user = KlayCredentials.create(ecKeyPair.getPrivateKey().toString(16), oldCredentials.getAddress());
+        chargeAccount(user.getAddress());
+        return user;
+    }
+
+    protected AccountKeyWeightedMultiSig createRandomAccountKeyWeightedMultiSig(List<ECKeyPair> ecKeyPairList) throws Exception {
+        Random random = new Random();
+        List<AccountKeyWeightedMultiSig.WeightedPublicKey> weightedTransactionPublicKeys = new ArrayList<>();
+        int sumOfWeight = 0;
+        for (ECKeyPair ecKeyPair : ecKeyPairList) {
+            int weight = random.nextInt(20) + 1;
+            sumOfWeight += weight;
+            AccountKeyWeightedMultiSig.WeightedPublicKey key = AccountKeyWeightedMultiSig.WeightedPublicKey.create(
+                    BigInteger.valueOf(weight),
+                    AccountKeyPublic.create(ecKeyPair.getPublicKey())
+            );
+            weightedTransactionPublicKeys.add(key);
+        }
+
+        AccountKeyWeightedMultiSig newAccountKey = AccountKeyWeightedMultiSig.create(
+                BigInteger.valueOf(random.nextInt(sumOfWeight) + 1),
+                weightedTransactionPublicKeys
+        );
+
+        return newAccountKey;
+    }
+
+    protected interface TransactionGetter {
+        TransactionTransformer get(String from) throws Exception;
+    }
+
+    protected interface ReceiptChecker {
+        void check(TransactionTransformer transactionTransformer, KlayTransactionReceipt.TransactionReceipt transactionReceipt) throws Exception;
+    }
+
+    class AccountSizeGenerator {
+        private int transactionAccountSize;
+        private int updateAccountSize;
+        private int feePayerAccountSize;
+
+        public AccountSizeGenerator() {
+            Random random = new Random();
+            this.transactionAccountSize = random.nextInt(9) + 1;
+            this.updateAccountSize = random.nextInt(9) + 1;
+            this.feePayerAccountSize = random.nextInt(9) + 1;
+        }
+
+        public int getTransactionAccountSize() {
+            return transactionAccountSize;
+        }
+
+        public int getUpdateAccountSize() {
+            return updateAccountSize;
+        }
+
+        public int getFeePayerAccountSize() {
+            return feePayerAccountSize;
+        }
+    }
+
+    public class RoleBasedAccountGenerator {
```

Comment:
This is because this test assumes a scenario where multiple users sign a transaction. For example, because there are multiple people who have a `transactionAccountCredential` and each of them can sign using their own `Credentials`, they are declared as a `List`.

Reply:
Oh I see. I mistake `KlayCredentials` for `ECKeyPair`.